controllers/btpoperator_controller.go

```diff
@@ -199,41 +207,65 @@ func (r *BtpOperatorReconciler) Reconcile(ctx context.Context, req ctrl.Request)
-			return ctrl.Result{}, r.HandleRedundantCR(ctx, oldestCr, cr)
+			return ctrl.Result{}, r.HandleRedundantCR(ctx, oldestCr, reconcileCr)
 		}
 	}
 
-	if ctrlutil.AddFinalizer(cr, deletionFinalizer) {
-		return ctrl.Result{}, r.Update(ctx, cr)
+	if ctrlutil.AddFinalizer(reconcileCr, deletionFinalizer) {
+		return ctrl.Result{}, r.Update(ctx, reconcileCr)
 	}
 
-	if !cr.ObjectMeta.DeletionTimestamp.IsZero() && cr.Status.State != v1alpha1.StateDeleting {
-		return ctrl.Result{}, r.UpdateBtpOperatorStatus(ctx, cr, v1alpha1.StateDeleting, conditions.HardDeleting, "BtpOperator is to be deleted")
+	if !reconcileCr.ObjectMeta.DeletionTimestamp.IsZero() && reconcileCr.Status.State != v1alpha1.StateDeleting {
+		return ctrl.Result{}, r.UpdateBtpOperatorStatus(ctx, reconcileCr, v1alpha1.StateDeleting, conditions.HardDeleting, "BtpOperator is to be deleted")
 	}
 
-	switch cr.Status.State {
+	switch reconcileCr.Status.State {
 	case "":
-		return ctrl.Result{}, r.HandleInitialState(ctx, cr)
+		return ctrl.Result{}, r.HandleInitialState(ctx, reconcileCr)
 	case v1alpha1.StateProcessing:
-		return ctrl.Result{RequeueAfter: ProcessingStateRequeueInterval}, r.HandleProcessingState(ctx, cr)
+		return ctrl.Result{RequeueAfter: ProcessingStateRequeueInterval}, r.HandleProcessingState(ctx, reconcileCr)
 	case v1alpha1.StateError, v1alpha1.StateWarning:
-		return ctrl.Result{}, r.HandleErrorState(ctx, cr)
+		return ctrl.Result{}, r.HandleErrorState(ctx, reconcileCr)
 	case v1alpha1.StateDeleting:
-		err := r.HandleDeletingState(ctx, cr)
-		if cr.IsReasonStringEqual(string(conditions.ServiceInstancesAndBindingsNotCleaned)) {
+		err := r.HandleDeletingState(ctx, reconcileCr)
+		if reconcileCr.IsReasonStringEqual(string(conditions.ServiceInstancesAndBindingsNotCleaned)) {
 			return ctrl.Result{RequeueAfter: ReadyStateRequeueInterval}, err
 		}
 		return ctrl.Result{}, err
 	case v1alpha1.StateReady:
-		return ctrl.Result{RequeueAfter: ReadyStateRequeueInterval}, r.HandleReadyState(ctx, cr)
+		return ctrl.Result{RequeueAfter: ReadyStateRequeueInterval}, r.HandleReadyState(ctx, reconcileCr)
 	}
 
 	return ctrl.Result{}, nil
 }
 
+func (r *BtpOperatorReconciler) setNewLeader(ctx context.Context, existingBtpOperators *v1alpha1.BtpOperatorList) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+	logger.Info(fmt.Sprintf("Found %d existing BtpOperators", len(existingBtpOperators.Items)))
+	oldestCr := r.getOldestCR(existingBtpOperators)
+	if oldestCr.GetStatus().State == v1alpha1.StateError && oldestCr.IsReasonStringEqual(string(conditions.OlderCRExists)) {
+		if err := r.UpdateBtpOperatorStatus(ctx, oldestCr, v1alpha1.StateProcessing, conditions.Processing,
+			fmt.Sprintf("%s is the new leader", oldestCr.GetName())); err != nil {
+			logger.Error(err, fmt.Sprintf("unable to set %s BtpOperator CR as the new leader", oldestCr.GetName()))
+			return ctrl.Result{}, err
+		}
+	}
+	logger.Info(fmt.Sprintf("%s BtpOperator is the new leader", oldestCr.GetName()))
```

Comment:
We reach this point not only when we successfully updated the oldest CR and set is as a new leader but even if `oldestCr.GetStatus().State == v1alpha1.StateError` but the reason is other than `conditions.OlderCRExists`.

Reply:
That's correct, the oldest BtpOperator will be picked up as a new leader and it doesn't have to be in Error State with OlderCRExists condition. If condition at [L249](https://github.com/kyma-project/btp-manager/pull/342/files#diff-c40c410af64e0ab33ab23c7fb7cd4bfdf0c7a4fe7c90b5d798a7938303c73edbR249) is required to avoid unnecessary, redundant reconciliations. BtpOperators with Statuses other than Error State, OlderCRExists Condition should be reconciled automatically.